plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/editor/ActiveEditorChangeListener.java

```diff
@@ -62,20 +67,33 @@ public void stop() {
 
     @Override
     public void partActivated(final IWorkbenchPartReference partRef) {
-        if (partRef.getPart(false) instanceof ITextEditor) {
-            ITextEditor editor = (ITextEditor) partRef.getPart(false);
-            handleEditorChange(editor);
+        Object part = partRef.getPart(false);
+        if (part instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) part);
+        } else if (part instanceof IEditorPart) {
+            // Check if this is an ADT editor by class name
+            String className = part.getClass().getName();
+            if (AbapUtil.isAdtEditor(className)) {
+                handleEditorChange((IEditorPart) part);
+            }
         }
     }
 
     @Override
     public void partClosed(final IWorkbenchPartReference partRef) {
-        if (partRef.getPart(false) instanceof ITextEditor) {
+        Object part = partRef.getPart(false);
+        if (part instanceof ITextEditor) {
             handleEditorChange(null);
+        } else if (part instanceof IEditorPart) {
+            // Check if this is an ADT editor by class name
+            String className = part.getClass().getName();
+            if (AbapUtil.isAdtEditor(className)) {
+                handleEditorChange(null);
+            }
```

Comment:
This can be simplified to, similar for L77. Consider passing in the part directory to `handleEditorChange(part)`
```
if (part instanceof ITextEditor || isAdtEditor(part)) {
    handleEditorChange(null)
}
```